src/ESP8266FS.java

```diff
@@ -216,6 +222,31 @@ private void createAndUpload(){
       return;
     }
 
+    // Find upload.py, don't fail if not present for backwards compat
+    File uploadPyFile = new File(platform.getFolder()+"/tools", "upload.py");
+    if (uploadPyFile.exists() && uploadPyFile.isFile()) {
+      uploadCmd = uploadPyFile.getAbsolutePath();
+    }
+    // Find python.exe if present, don't fail if not found for backwards compat
+    String toolPyCmd = pythonCmd;
+    if ((toolPyCmd != null ) && !toolPyCmd.isEmpty()) {
+      File toolPyFile = new File(platform.getFolder()+"/tools", toolPyCmd);
```

Comment:
This three line block seems to be repeating multiple times for different file names. Will it help to wrap it into a function?

Reply:
Replaced the nested ifs with a simple for(String[]) iterator, fewer code lines and no repetition now.